src/php/Runtime.php

```diff
@@ -11,63 +13,46 @@
 
 class Runtime {
 
-    /**
-     * @var GlobalEnvironment
-     */
-    private $globalEnv;
+    private GlobalEnvironment $globalEnv;
 
-    /**
-     * @var string[]
-     */
-    private $loadedNs = [];
+    /** @var string[] */
+    private array $loadedNs = [];
 
-    /**
-     * @var array
-     */
-    private $paths = [];
+    private array $paths = [];
 
-    /**
-     * @var string|null
-     */
-    private $cacheDiretory;
+    private ?string $cacheDirectory = null;
 
-    /**
-     * @var Runtime|null
-     */
-    private static $instance;
+    private static ?Runtime $instance = null;
 
-    private function __construct(GlobalEnvironment $globalEnv = null, string $cacheDiretory = null) {
+    private function __construct(GlobalEnvironment $globalEnv = null, string $cacheDirectory = null) {
         set_exception_handler(array($this, 'exceptionHandler'));
 
         if (is_null($globalEnv)) {
             $globalEnv = new GlobalEnvironment();
         }
         $this->globalEnv = $globalEnv;
-        $this->cacheDiretory = $cacheDiretory;
+        $this->cacheDirectory = $cacheDirectory;
         $this->addPath('phel\\', [__DIR__ . '/../phel']);
     }
 
-    public static function initialize(GlobalEnvironment $globalEnv = null, string $cacheDiretory = null) {
+    public static function initialize(GlobalEnvironment $globalEnv = null, string $cacheDirectory = null) {
         if (self::$instance !== null) {
             throw new Exception("Runtime is already initialized");
         }
 
-        self::$instance = new Runtime($globalEnv, $cacheDiretory);
+        self::$instance = new Runtime($globalEnv, $cacheDirectory);
         return self::$instance;
     }
 
     /**
      * @interal
      */
-    public static function newInstance(GlobalEnvironment $globalEnv = null, string $cacheDiretory = null) {
-        return new Runtime($globalEnv, $cacheDiretory);
+    public static function newInstance(GlobalEnvironment $globalEnv = null, string $cacheDirectory = null): self {
+        return new Runtime($globalEnv, $cacheDirectory);
     }
 
-    /**
-     * @param Exception $exception
-     */
-    public function exceptionHandler($exception) {
-        if (php_sapi_name() == 'cli') {
+    public function exceptionHandler(Exception $exception): void {
```

Comment:
Actually `Exception` is wrong. It must be `Throwable`. See https://www.php.net/manual/en/function.set-exception-handler.php

Reply:
Done